src/config/datasource.ts

```diff
@@ -1,25 +1,31 @@
-import { DataSource, DataSourceOptions } from "typeorm";
+import { DataSource, DataSourceOptions } from 'typeorm';
 import { ConfigModule, ConfigService } from '@nestjs/config';
 
 ConfigModule.forRoot({
-    isGlobal: true,
-    envFilePath: [`.env`, `${process.env.NODE_ENV}.env`],
+  isGlobal: true,
+  envFilePath: [`.env`, `${process.env.NODE_ENV}.env`],
 });
 
 const configService = new ConfigService();
 
 export const DataSourceConfig: DataSourceOptions = {
-    type: 'mysql',
-    host: configService.get('DB_HOST'),
-    port : configService.get('PORT'),
-    username: configService.get('DB_USER'),
-    password: configService.get('DB_PASSWORD'),
-    database: configService.get('DB_NAME'),
-    entities: ['dist/**/*.entity.js'],
-    migrations: ['dist/../migrations/*{.ts,.js}'],
-    synchronize: false,
-    migrationsRun: true,
-    logging: false,
-}
+  type: configService.get('DB_TYPE') as any,
+  host: configService.get('DB_HOST') as string,
+  port: configService.get('PORT') as number,
+  username: configService.get('DB_USER') as string,
+  password: configService.get('DB_PASSWORD') as string,
+  database: configService.get('DB_NAME') as string,
```

Comment:
Una forma mas elegante de convertir a ese tipo de dato y asegurarse de que lo sea
```suggestion
  host: configService.get('DB_HOST') + '',
  port: +configService.get('PORT'),
  username: configService.get('DB_USER') + '',
  password: configService.get('DB_PASSWORD') + '',
  database: configService.get('DB_NAME') + '',
```